hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/AbstractFindTargetGreedy.java

```diff
@@ -162,9 +165,16 @@ private boolean containerMoveSatisfiesPlacementPolicy(
             .filter(datanodeDetails -> !datanodeDetails.equals(source))
             .collect(Collectors.toList());
     replicaList.add(target);
-    ContainerPlacementStatus placementStatus =
-        placementPolicy.validateContainerPlacement(replicaList,
-        containerInfo.getReplicationConfig().getRequiredNodes());
+    ContainerPlacementStatus placementStatus;
+    switch (containerInfo.getReplicationType()) {
+    case EC:
+      placementStatus = ecPlacementPolicy.validateContainerPlacement(
+          replicaList, containerInfo.getReplicationConfig().getRequiredNodes());
+      break;
+    default:
+      placementStatus = placementPolicy.validateContainerPlacement(replicaList,
+          containerInfo.getReplicationConfig().getRequiredNodes());
+    }
```

Comment:
Instead of deciding which policy to use in ContainerBalancer, would it be better if we decide this in the SCM or someplace else? That way we won't have to keep on adding a new switch case here if more policies are introduced in the future.